iot/iot-infinispan-base/src/main/java/io/enmasse/iot/infinispan/cache/AbstractCacheProvider.java

```diff
@@ -82,6 +83,25 @@ public void close() throws Exception {
         stop();
     }
 
+    protected void uploadSchema(final String fileName, final String schema) {
+
+        if (!this.properties.isUploadSchema()) {
+            return;
+        }
+
+        var cache = this.remoteCacheManager
+                .getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
+
+        if (this.properties.isOverrideSchema()) {
+            log.info("Uploading (put) schema - name: {}, schema: {}", fileName, schema);
+            cache.put(fileName, schema);
```

Comment:
It'd be good to log the result of the operations for the easier debugging (as the future improvement)